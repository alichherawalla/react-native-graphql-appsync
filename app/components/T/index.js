/**
 *
 * T
 *
 */

import React from 'react'
import PropTypes from 'prop-types'
import { compose } from 'redux'
import { injectIntl } from 'react-intl'
import { Text } from 'react-native-paper'
import { StyleSheet } from 'react-native'
import { rnFontSizeAdapter, rnFontWeightAdapter } from 'app/utils'

const T = ({ intl, id, values, text, size = 'small', weight = 'normal' }) => {
  const styles = StyleSheet.create({
    text: {
      ...rnFontSizeAdapter(size),
      ...rnFontWeightAdapter(weight)
    }
  })
  text = text || intl.formatMessage({ id }, { ...values })
  return (
    <Text testID="t" style={styles.text}>
      {text}
    </Text>
  )
}

T.propTypes = {
  id: PropTypes.string,
  intl: PropTypes.object,
  values: PropTypes.object,
  size: PropTypes.string,
  weight: PropTypes.string,
  text: PropTypes.string
}

export default compose(injectIntl)(T)
