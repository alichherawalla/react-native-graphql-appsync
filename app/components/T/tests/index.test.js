/**
 *
 * Tests for T
 *
 * @see https://github.com/react-boilerplate/react-boilerplate/tree/master/docs/testing
 *
 */

/* global shallowWithIntl */
/* eslint no-undef: "error" */
import React from 'react'
import Enzyme from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'

import T from '../index'

Enzyme.configure({ adapter: new Adapter() })

describe('<T />', () => {
  it('Should render and match the snapshot', () => {
    const component = shallowWithIntl(<T id="test" />)
    expect(component).toMatchSnapshot()
  })
})
