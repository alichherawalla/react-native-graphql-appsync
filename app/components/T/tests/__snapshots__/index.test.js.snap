// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<T /> Should render and match the snapshot 1`] = `ShallowWrapper {}`;
